adapters/yieldlab/yieldlab.go

```diff
@@ -0,0 +1,309 @@
+			return "", "", fmt.Errorf("failed to parse ExtRegs in Yieldlab GDPR check: %v", err)
+		}
+		if extRegs.GDPR != nil && (*extRegs.GDPR == 0 || *extRegs.GDPR == 1) {
+			gdpr = strconv.Itoa(int(*extRegs.GDPR))
+		}
+	}
+
+	consent := ""
+	if request.User != nil && request.User.Ext != nil {
+		var extUser openrtb_ext.ExtUser
+		if err := json.Unmarshal(request.User.Ext, &extUser); err != nil {
+			return "", "", fmt.Errorf("failed to parse ExtUser in Yieldlab GDPR check: %v", err)
+		}
+		consent = extUser.Consent
+	}
+
+	return gdpr, consent, nil
+}
+
+func (a *YieldlabAdapter) makeTargetingValues(params *openrtb_ext.ExtImpYieldlab) string {
+	values := url.Values{}
+	for k, v := range params.Targeting {
+		values.Set(k, v)
+	}
+	return values.Encode()
+}
+
+func (a *YieldlabAdapter) MakeRequests(request *openrtb.BidRequest, _ *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, []error{fmt.Errorf("invalid request %+v, no Impressions given", request)}
+	}
+
+	bidURL, err := a.makeEndpointURL(request, a.mergeParams(a.parseRequest(request)))
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	headers := http.Header{}
+	headers.Add("Accept", "application/json")
+	if request.Site != nil {
+		headers.Add("Referer", request.Site.Page)
+	}
+	if request.Device != nil {
+		headers.Add("User-Agent", request.Device.UA)
+		headers.Add("X-Forwarded-For", request.Device.IP)
+	}
+	if request.User != nil {
+		headers.Add("Cookie", "id="+request.User.BuyerUID)
+	}
+
+	return []*adapters.RequestData{{
+		Method:  "GET",
+		Uri:     bidURL,
+		Headers: headers,
+	}}, nil
+}
+
+// parseRequest extracts the Yieldlab request information from the request
+func (a *YieldlabAdapter) parseRequest(request *openrtb.BidRequest) []*openrtb_ext.ExtImpYieldlab {
+	params := make([]*openrtb_ext.ExtImpYieldlab, 0)
+
+	for i := 0; i < len(request.Imp); i++ {
+		bidderExt := new(adapters.ExtImpBidder)
+		if err := json.Unmarshal(request.Imp[i].Ext, bidderExt); err != nil {
+			continue
+		}
+
+		yieldlabExt := new(openrtb_ext.ExtImpYieldlab)
+		if err := json.Unmarshal(bidderExt.Bidder, yieldlabExt); err != nil {
+			continue
+		}
+
+		params = append(params, yieldlabExt)
+	}
+
+	return params
+}
+
+func (a *YieldlabAdapter) mergeParams(params []*openrtb_ext.ExtImpYieldlab) *openrtb_ext.ExtImpYieldlab {
+	var adSlotIds []string
+	targeting := make(map[string]string)
+
+	for _, p := range params {
+		adSlotIds = append(adSlotIds, p.AdslotID)
+		for k, v := range p.Targeting {
+			targeting[k] = v
+		}
+	}
+
+	return &openrtb_ext.ExtImpYieldlab{
+		AdslotID:  strings.Join(adSlotIds, adSlotIdSeparator),
+		Targeting: targeting,
+	}
+}
+
+// MakeBids make the bids for the bid response.
+func (a *YieldlabAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode != 200 {
```

Comment:
Some bidders return a 204 when not making a bid. I just want to confirm Yieldlab doesn't have that behavior.